net/src/ipc_net_worker.rs

```diff
@@ -232,13 +182,87 @@ impl IpcNetWorker {
             ipc_relay,
             ipc_relay_receiver,
             is_ready: false,
-            state: "undefined".to_string(),
+            last_known_state: "undefined".to_string(),
             last_state_millis: 0.0_f64,
             bootstrap_nodes,
             endpoint,
         })
     }
+}
+
+impl NetWorker for IpcNetWorker {
+    /// stop the net worker
+    fn stop(self: Box<Self>) -> NetResult<()> {
+        self.ipc_relay.stop()?;
+        Ok(())
+    }
 
+    /// we got a message from holochain core
+    /// (just forwards to the internal worker relay)
+    fn receive(&mut self, data: Protocol) -> NetResult<()> {
+        self.ipc_relay.send(data)?;
+        Ok(())
+    }
+
+    /// do some upkeep on the internal worker
+    /// IPC server state handling / magic
+    fn tick(&mut self) -> NetResult<bool> {
+        let mut has_done_something = false;
+
+        // WTF?
```

Comment:
WTF? : ) lol

could say something like

```rust
// if we have yet to issue a state request to the networking process, issue one now
```

Reply:
oops, forgot to remove that one :p